test/unit/test_visual_recognition_v4.py

```diff
@@ -12,824 +13,1200 @@
+    @responses.activate
+    def test_get_training_usage_response(self):
+        body = self.construct_full_body()
+        response = fake_response_TrainingEvents_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 2: Send only required fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_get_training_usage_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response_TrainingEvents_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_get_training_usage_empty(self):
+        check_empty_response(self)
+        assert len(responses.calls) == 1
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v4/training_usage'
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.GET,
+                      url,
+                      body=json.dumps(response),
+                      status=200,
+                      content_type='application/json')
+
+    def call_service(self, body):
+        service = VisualRecognitionV4(authenticator=NoAuthAuthenticator(),
+                                      version='2019-02-11')
         service.set_service_url(base_url)
+        output = service.get_training_usage(**body)
+        return output
 
-        detailed_response = service.add_image_training_data(
-            collection_id='collection_id', image_id='image_id')
-        result = detailed_response.get_result()
-        assert result is not None
-        assert len(responses.calls) == 2
+    def construct_full_body(self):
+        body = dict()
+        body['start_time'] = "string1"
+        body['end_time'] = "string1"
+        return body
 
-    #########################
-    # userData
-    #########################
+    def construct_required_body(self):
+        body = dict()
+        return body
 
+
+# endregion
+##############################################################################
+# End of Service: Training
+##############################################################################
+
+##############################################################################
+# Start of Service: UserData
+##############################################################################
+# region
+
+
+#-----------------------------------------------------------------------------
+# Test Class for delete_user_data
+#--------------
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'e' is assigned to but never used](https://app.codacy.com/app/watson-developer-cloud/python-sdk/pullRequest?prid=4546453)